examples/src/components/CopyLink.tsx

```diff
@@ -0,0 +1,33 @@
+'use client'
+import { useState } from 'react'
+
+export default function CopyLink() {
+  const [buttonState, setButtonState] = useState('Copy link')
+
+  const copyLinkToClipboard = () => {
+    const currentPageURL = window.location.href
+
+    navigator.clipboard
+      .writeText(currentPageURL)
+      .then(() => {
+        setButtonState('Copied!')
+      })
+      .catch((error) => {
+        console.error('Failed to copy link: ', error)
+      })
+  }
+
+  return (
+    <div className="p-6 flex flex-col text-left rounded-lg bg-white/50 border border-white/80 shadow-sm">
+      <span className="text-pink-900 pb-2">
+        Open this doc in a new window or share with a friend:
+      </span>
+      <button
+        className="w-28 py-2 rounded-lg bg-pink-950 text-white transition-all hover:bg-pink-900"
+        onClick={copyLinkToClipboard}
+      >
+        {buttonState}
+      </button>
+    </div>
```

Comment:
I like this idea a lot, but I kinda feel like it visually competes with the demos themselves too much:

<img width="1351" alt="Screenshot 2023-08-04 at 11 19 21 AM" src="https://github.com/drifting-in-space/y-sweet/assets/2099249/8e537dc7-e479-4bfc-9c3e-3d8f49372ec4">

I played around with making it smaller like this:

<img width="714" alt="Screenshot 2023-08-04 at 11 21 52 AM" src="https://github.com/drifting-in-space/y-sweet/assets/2099249/33e41d2e-6e57-407c-8b34-d27d5ea458d2">

I also tried moving it up next to the title, but it still didn't look right to me.

I like what Liveblocks does by placing it in their top bar:

![liveblocks-share](https://github.com/drifting-in-space/y-sweet/assets/2099249/7042f5bd-0539-4772-bf7f-b722870d5f87)

Any thoughts?

Reply:
i like these approaches. i can work on the simpler inline version for now to get this out sooner (and follow up with something more sophisticated if needed)